### ISSUE TITLE
Add limit commanders modoption

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K#1844]
This modoption limits the number of commanders spawned per allyTeam. Beamer wants it for playing 1v1 with commshare.

**Comment by ArchShaman** [comment]
I suggest having it be a commshare option only:
* Without all mode, who gets a com and who doesn't?
* What to do with no com players?
* If the answer is to merge them, merge them with who?

This is pretty easily done I think. Simply remove the 'mintime' part of the config if 'team 1v1' is enabled. Or something.

**Comment by GoogleFrog** [comment]
Yes, only enabled during all merge would be good. The question of who gets the comm is still relevant because they need to select their commander.

**Comment by ArchShaman** [comment]
All mode takes the lowest playerID in each of the allyteams.

**Comment by GoogleFrog** [comment]
I would rather have the commander limit be a number modoption. This lets you play, say, a 2v2 with 5 players.

**Comment by ArchShaman** [comment]
Idea:
* Take the first X players, let them have a commander and count them towards the 'waiting for' people in startpos gadget.
* Merge the non-commander players into these players evenly at game start. Start with the lowest human teamID and divide them equally (as possible). Possibly don't count them in lagmonitor?

**Comment by GoogleFrog** [comment]
I think the first X players to choose a commander should get their commander. The later players are sent an event to tell the UI to hide commander screen and remove their ability/need to place a startpoint. My imagined scenario for this is highly organized teams so they are not going to be fighting over placement. It's important to give these teams the option as to which commanders are used.

**Comment by ArchShaman** [comment]
Perhaps also have a "Commanders remaining: " .. modoptions["CommanderLimit"] - currentlySelectedCommanders displayed inbetween Waiting for players and "Type !force to start sooner."
